scripts/zkai_native_attention_mlp_rmsnorm_adjacent_layout_gate.py

```
@@ -0,0 +1,581 @@
+        if path in by_path:
+            raise AdjacentLayoutGateError(f"duplicate accounting row path: {path}")
+        by_path[path] = row
+    expected_paths = {entry["path"] for entry in EXPECTED_VARIANTS.values()}
+    if set(by_path) != expected_paths:
+        raise AdjacentLayoutGateError("accounting row inventory drift")
+    return by_path
+
+
+def build_payload(*, include_mutations: bool = True) -> dict[str, Any]:
+    if sha256_file(ACCOUNTING_PATH) != EXPECTED_ACCOUNTING_SHA256:
+        raise AdjacentLayoutGateError("adjacent accounting source digest drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)